src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator_entrypoint.py

```diff
@@ -207,8 +207,39 @@ def run_step_on_kubernetes(step_name: str) -> None:
             or settings.service_account_name,
             mount_local_stores=mount_local_stores,
             owner_references=owner_references,
+            namespace=args.kubernetes_namespace,
         )
 
+        # Step pods should reuse secrets created by the orchestrator pod
+        # Only create secrets if they don't already exist
+        for secret_manifest in secret_manifests:
+            secret_name = secret_manifest["metadata"]["name"]
+            try:
+                # Check if secret already exists
+                core_api.read_namespaced_secret(
+                    name=secret_name, namespace=args.kubernetes_namespace
+                )
+                logger.debug(
+                    f"imagePullSecret {secret_name} already exists, reusing it"
+                )
+            except k8s_client.rest.ApiException as e:
+                if e.status == 404:
+                    # Secret doesn't exist, create it
+                    try:
+                        kube_utils.create_or_update_secret_from_manifest(
+                            core_api=core_api,
+                            secret_manifest=secret_manifest,
+                        )
+                        logger.debug(f"Created imagePullSecret {secret_name}")
+                    except Exception as create_e:
+                        logger.warning(
+                            f"Failed to create imagePullSecret {secret_name}: {create_e}"
+                        )
+                else:
+                    logger.warning(
+                        f"Failed to check for existing secret {secret_name}: {e}"
+                    )
```

Comment:
If I understand it correctly, this secret never gets cleaned up? We're already using a kubernetes secret for the ZenML token which already has some cleanup logic, it probably makes sense to also apply this here?